CMSgov/design-system
# [WNMGDS-848] Replace React.HTMLProps with React.ComponentPropsWithRef

## Summary
This PR is for replacing instances of React.HTMLProps in TextField, Dropdown, and in any other components that mirror HTML elements

https://react-typescript-cheatsheet.netlify.app/docs/advanced/patterns_by_usecase/#wrappingmirroring-a-html-element
We want to follow that reference for how to wrap or mirror components in our typescript definitions. Currently we are using React.HTMLProps, which is not recommended.

### Changed
Changed all instances of React.HTMLProps to use React.ComponentPropsWithRef

## How to test
Pull components into creact-react-app and create-react-app-typescript and make sure there are no errors.
